api/v2/serializers/details/resource_request_actions.py

```diff
@@ -39,17 +39,20 @@ def create(self, validated_data):
         user = validated_data.get('approved_by')
         resource_request = validated_data.get('resource_request')
         identity = validated_data.get('identity')
-        data = {
-            'identity': identity.id,
-            'resource_request': resource_request.id,
-            'approved_by': user.username,
-            'quota': quota.id
-        }
-        event_serializer = QuotaAssignedByResourceRequestSerializer(data=data)
-        event_serializer.is_valid(raise_exception=True)
 
-        # Synchronous call to EventTable -> Set Quota for Identity's CloudProvider -> Save the Quota to Identity
-        event_serializer.save()
+        # This api is designed to update a single identity's quota, instead we
+        # update all identities for the user. This syncs the quota across all of
+        # the user's active identities.
+        for ident in [i for i in user.identity_set.all() if i.is_active()]:
```

Comment:
Depending on the length of `user.identity_set.all()` this could be a long/expensive call. 

Moving this entire block into an AsyncTask wouldn't be unreasonable.

Reply:
Yeah we will need to do that.